src/main/java/org/ieeervce/gatekeeper/controller/RequestFormController.java

```diff
@@ -0,0 +1,105 @@
+package org.ieeervce.gatekeeper.controller;
+
+import jakarta.mail.Multipart;
+import org.ieeervce.gatekeeper.InvalidDataException;
+import org.ieeervce.gatekeeper.ItemNotFoundException;
+import org.ieeervce.gatekeeper.PDFNotConversionException;
+import org.ieeervce.gatekeeper.dto.RequestDTO;
+import org.ieeervce.gatekeeper.entity.*;
+
+import org.ieeervce.gatekeeper.service.RequestFormService;
+import org.ieeervce.gatekeeper.service.RoleService;
+import org.ieeervce.gatekeeper.service.UserService;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import java.util.Optional;
+
+import java.awt.*;
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+
+import static org.ieeervce.gatekeeper.config.SecurityConfiguration.getRequesterDetails;
+
+@RestController
+@RequestMapping("/requestForm")
+public class RequestFormController {
+
+    private final ModelMapper modelMapper;
+    private final RequestFormService requestFormService;
+
+    private final RoleService roleService;
+    private final UserService userService;
+    RequestFormController(RequestFormService requestFormService, ModelMapper modelMapper, UserService userService, RoleService roleService){
+        this.requestFormService = requestFormService;
+        this.modelMapper= modelMapper;
+        this.userService = userService;
+        this.roleService=roleService;
+    }
+
+    @GetMapping
+    public List<RequestForm> getAll(){
+        return requestFormService.list();
+    }
+
+    @GetMapping("/byRequester")
+    public List<RequestForm> getByUser()
+    {
+        return requestFormService.getRequestFormByRequester(userService.getUserByEmail(getRequesterDetails()).get());
```

Comment:
Would `@AuthenticationPrincipal` be a good way to get this data you need?

Reply:
Will check that out,thanks!